app/concerns/sign_in_with_token.rb

```diff
@@ -4,13 +4,26 @@ module SignInWithToken
   included do
     validates :sign_in_token, uniqueness: true, allow_nil: true
 
-    def generate_token!
-      update!(sign_in_token: SecureRandom.uuid)
+    def generate_token!(ttl: nil)
+      ttl ||= (ENV['SIGN_IN_TOKEN_TTL_SECONDS'] || 1800)
```

Comment:
Can we please pull out 1800 as a constant at the top of the file? it's a bit buried here.

Reply:
Yep, fair point - I've done that and also extracted the calculation of effective TTL out to a separate method